lib/devise/models/authenticatable.rb

```diff
@@ -253,6 +253,7 @@ def find_or_initialize_with_error_by(attribute, value, error=:invalid) #:nodoc:
 
         # Find or initialize a record with group of attributes based on a list of required attributes.
         def find_or_initialize_with_errors(required_attributes, attributes, error=:invalid) #:nodoc:
+          attributes = attributes.to_unsafe_h.with_indifferent_access if attributes.respond_to? :to_unsafe_h
```

Comment:
I'll try to debug this one out and see if we can avoid to pass a `Parameters` object down the models in the builtin controllers - now that they aren't inheriting from `Hash` anymore there are a few subtle API differences that can be pretty annoying to debug and we could save the developers from having to deal with that in their apps.